omnibus editorial grab-bag, of things identified and lightly-discussed on today's call (#12)
decentralized-identity/waci-didcomm: spec/spec.md

## Abstract

> There are interactions between a "holder"/subject and an "issuer"/identity provider,

Reviewer: ```suggestion
There are interactions between a "holder" and an "issuer",
```

> or between a "holder"/subject and a "verifier"/relying party, that require passing 

Reviewer: ```suggestion
or between a "holder" and a "verifier" that require passing 
```

[Verifiable Credentials](https://www.w3.org/TR/vc-data-model/) that make use of
[BBS+ LD-Signatures](https://w3c-ccg.github.io/ldp-bbs2020).

## Status of This Document

> [WACI DRAFT](https://identity.foundation/wallet-and-credential-interactions/)

Reviewer 2: This link seems wrong to me, is it correct?

Reviewer 2: And should it be `WACI DIDComm` rather than `WACI DRAFT` ?

Reviewer: No, we are attempting to cite the "original" waci.

Reviewer 2: I think it is important to cite the original WACI paper.
I think the 'Status of this Document' section is probably the wrong place to do so.

Author: Wow sorry, got my wires crossed, my mistake. I think Brent's right, this link made more sense with it pointed to the rendered spec in case the reader is reading this anywhere else (github, aggregators, who knows). We link to the WACI a few lines above in the dependency list, which makes sense.

> [WACI v0.1.0](https://identity.foundation/wallet-and-credential-interactions/)

Reviewer: ```suggestion
[WACI DRAFT](https://identity.foundation/wallet-and-credential-interactions/)
```

`Presentation Definition` data object which may be used by a verifier (i.e., a 
"relying party" in OIDC parlance) to request information, and a `Presentation 
Submission` object which describes the relationship between the `Presentation 
Definition` and the submitted verifiable information.

Since
[Presentation Exchange](https://identity.foundation/presentation-exchange/spec/v1.0.0/)
supports the use of `Verifiable Presentations` as a response to a `Presentation
Definition`, while remaining agnostic to transport protocols, it is ideal for
our purposes.

### Protocol Considerations
Before fixing the scope of this work item and settling on using elements of
[DIDComm v2.0](https://identity.foundation/didcomm-messaging/spec/) to securely
communicate the data objects, there was a long and lively discussion about other
options at IIW. 

- [CHAPI](https://w3c-ccg.github.io/credential-handler-api/) is an API for
exchanging data between websites and browsers. It was seen as too limited to a
particular technology to be widely useful for wallet applications on smart
phones and other devices without significant changes.

> - [VC-API](https://w3c-ccg.github.io/vc-api/) (fka VC-HTTP-API) was seen as promising, 

Reviewer: I will wait for this to be merged before taking a swing at this: https://github.com/decentralized-identity/waci-didcomm/issues/14

Author: don't hurt me

describes a `Verifiable Presentation` object that is designed for sharing
information from `Verifiable Credentials`. 

### Credential Manifest
In issuance situations, the credentials on offer by a given issuer can be
defined in objects that can be passed back and forth, in the "Presentation 
Exchange" data model/abstraction layering. These issuance-definition 

> objects are defined for issuance flows and wallet-issuer/wallet-verifier, a "sister specification" that spun out of the Presentation Exchange work-item group defines those objects: [Credential Manifest](https://identity.foundation/credential-manifest/#credential-manifest-2), also a work item of the DIF Claims and Credentials WG.

Reviewer 2: ```suggestion
objects are defined for issuance flows. A "sister specification" of the Presentation Exchange work-item group defines those objects: [Credential Manifest](https://identity.foundation/credential-manifest/#credential-manifest-2).
```

a method for securely communicating authenticated messages between entities that
control [Decentralized Identifiers](https://www.w3.org/TR/did-core/) along any
transport layer. It allows for two parties to mutually authenticate and securely
communicate. DIDComm has a large community interested in using it, and many have
already implemented DIDComm v1.0. 

- [OIDC-SIOP](https://openid.bitbucket.io/connect/openid-connect-self-issued-v2-1_0.html)
seeks to bridge existing federated identity capabilities on the internet with
principles of self-sovereign identity, and may ultimately be a good fit for this
protocol. It was not selected for this version of the specification primarily
because v1 of the specification has been deprecated, and v2 of the specification
was somewhat unstable at the time.

- [Aries Protocols](https://github.com/hyperledger/aries-rfcs), specifically the current stable drafts of
[present proof protocol V2](https://github.com/hyperledger/aries-rfcs/tree/master/features/0454-present-proof-v2)
and [issue credential V2](https://github.com/hyperledger/aries-rfcs/tree/main/features/0453-issue-credential-v2).

> An [unstable draft of V3](https://github.com/decentralized-identity/waci-didcomm/blob/main/present_proof/present-proof-v3.md)

Reviewer 2: ```suggestion
A [draft of V3](https://github.com/decentralized-identity/waci-didcomm/blob/main/present_proof/present-proof-v3.md)
```
